File: NearestNeighbor.py
from PIL import Image, ImageDraw
class NearestNeighbor:
    #convert the bitmap image into an object matrix
    @staticmethod
    def create_matrix(image):
        width, height = image.size
        matrix = []
        for i in range(width):
            matrix.append([])
            for j in range(height):
                r, g, b = image.getpixel((i, j))
                color = (r, g, b)
                matrix[i].append(color)
        return matrix
    #redimension the object matrix, it can be used to other type of data instead of image
    @staticmethod
    def redimension(matrix, new_width, new_height):
        new_matrix = []
        delta_x = float(new_width) / len(matrix)
        delta_y = float(new_height) / len(matrix[0])
        for i in range(new_width):
            new_matrix.append([])
            for j in range(new_height):
                x = int(i / delta_x)
                y = int(j / delta_y)
                new_matrix[i].append(matrix[x][y])
        return new_matrix
    #creates an image from a object matrix
    @staticmethod
    def create_image(matrix):
        image = Image.new('RGB', (len(matrix), len(matrix[0])), color='white')
        for i in range(len(matrix)):
            for j in range(len(matrix[0])):
                image.putpixel((i, j), matrix[i][j])
        return image

File: nearest_neighbor.php
<?php
class NearestNeighbor
{
    #convert the bitmap image into an object matrix
    public static function create_matrix($file_name)
    {
        $matrix = array();
        list($width, $height, $type, $attr) = getimagesize($file_name);
        if (strpos($file_name, "jpg") !== false) {
            $image = imagecreatefromjpeg($file_name);
        } else if (strpos($file_name, "png") !== false) {
            $image = imagecreatefrompng($file_name);
        } else {
            throw new Exception("File not supported.");
        }
        for ($i = 0; $i < $width; $i++) {
            array_push($matrix, array());
            for ($j = 0; $j < $height; $j++) {
                $rgb = imagecolorat($image, $i, $j);
                $red = ($rgb >> 16) & 255;
                $green = ($rgb >> 8) & 255;
                $blue = $rgb & 255;
                array_push($matrix[count($matrix) - 1], array("red" => $red, "green" => $green, "blue" => $blue));
            }
        }
        return $matrix;
    }
    #redimension the object matrix, it can be used to other type of data instead of image
    public static function redimension($matrix, $new_width, $new_height)
    {
        $new_matrix = array();
        $delta_x = $new_width / count($matrix);
        $delta_y = $new_height / count($matrix[0]);
        for ($i = 0; $i < $new_width; $i++) {
            array_push($new_matrix, array());
            for ($j = 0; $j < $new_height; $j++) {
                $x = floor($i / $delta_x);
                $y = floor($j / $delta_y);
                array_push($new_matrix[count($new_matrix) - 1], $matrix[$x][$y]);
            }
        }
        return $new_matrix;
    }
    #Creates an image from a object matrix
    public static function create_image($matrix)
    {
        $image = imagecreatetruecolor(count($matrix), count($matrix[0]));
        for ($i = 0; $i < count($matrix); $i++) {
            for ($j = 0; $j < count($matrix[0]); $j++) {              
                $color = imagecolorallocate($image, $matrix[$i][$j]["red"], $matrix[$i][$j]["green"], $matrix[$i][$j]["blue"]);
                imagesetpixel($image, $i, $j, $color);
            }
        }
        header('Content-Type: image/png');
        imagepng($image);
    }
}